Hook and callback changes

ilessiivi opened the pull request:
* Adds the same warning about filtering in file-added to aHook than in the synchronous hook and fixes a typo in it
* Fixes calling emitCatchAll in aHook when there's no user defined callback
* Fixes a typo with Array.prototype.includes in aHook
* Fixes inverted logic with filter-file's callback in aHook
* Styling: changed let and var to const

drzraf commented:
Quoting the README:
> `filter-file(<FlowFile> file, event) : boolean` The boolean return value decide whether this particular file must be processed or ignored

It's the only one checking the return value.
This change implies that if multiple hooks are attached, then one `true` would keep the file whereas the previous behavior was that one `false` would skip the file.

It's clearly an edge-case, but I still think that `&` the boolean values makes more sense so that `false` wins.
(@AidasK )

AidasK commented:
I am not sure about this one either. Do we really need to invert this place? Could you give some examples why you think it's a bad behavior?

drzraf commented on src/Eventizer.js:
```diff
@@ -347,17 +347,22 @@ export default class extends EventTarget {
     this.emitCatchAll(name, ...args);
-    return isFilter ? returns.include(false) : returns;
+
+    if (name === 'file-added' && returns.includes(false)) {
```
`aHook` is only going to be called by user of `async` callback. So, contrary to `hook`, warning is not necessary.

ilessiivi replied:
Yeah, that can't be met accidentally by just upgrading from v2. Removed the extra warning from `aHook` (but will probably need it in the async hook again if they're combined into one later.)

ilessiivi commented:
The filter logic in the squashed commit is now checking that none of the hooks return false (I missed that this was the original behavior - sorry), and I added a new test to check for it in the async code path.

The second warning is now removed, though it'll be needed in the async hook later if the sync/async code paths get combined.

drzraf commented:
Amazing!
